internal/guest/runtime/hcsv2/uvm.go

```diff
@@ -435,6 +436,50 @@ func (*Host) Shutdown() {
 	_ = syscall.Reboot(syscall.LINUX_REBOOT_CMD_POWER_OFF)
 }
 
+// Called to shutdown a container
+func (h *Host) ShutdownContainer(ctx context.Context, containerID string, graceful bool) error {
+	c, err := h.GetCreatedContainer(containerID)
+	if err != nil {
+		return err
+	}
+
+	err = h.securityPolicyEnforcer.EnforceShutdownContainerPolicy(containerID)
+	if err != nil {
+		return err
+	}
+
+	signal := unix.SIGTERM
+	if !graceful {
+		signal = unix.SIGKILL
+	}
+
+	return c.Kill(ctx, signal)
+}
+
+func (h *Host) SignalContainerProcess(ctx context.Context, containerID string, processID uint32, signal syscall.Signal) error {
+	c, err := h.GetCreatedContainer(containerID)
+	if err != nil {
+		return err
+	}
+
+	// Don't allow signalProcessV2 to route around container shutdown policy
+	// Sending SIGTERM or SIGKILL to a containers init process will shut down
+	// the container.
+	if processID == c.initProcess.pid {
```

Comment:
shouldn't we reject this altogether, i.e. signal process shouldn't be used signal the container's init process.

Reply:
reading more through the code, I don't think we need to handle this case. Container's init process isn't tracked by `container.processes`, so the lookup on L472 will fail if we pass the container init process pid. With this change, we're actually introducing a new behavior.

Reply:
As discussed, I've removed this if clause.